cub/test/catch2_test_block_run_length_decode.cu

```diff
@@ -308,11 +316,12 @@ void TestAlgorithmSpecialisation()
 
   using RunItemT      = float;
   using RunLengthT    = uint32_t;
-  using ItemItT       = thrust::counting_iterator<RunItemT>;
-  using RunLengthsItT = thrust::transform_iterator<ModOp, thrust::counting_iterator<RunLengthT>>;
+  using CasterOp      = CastOp<RunItemT>;
+  using ItemItT       = cuda::transform_iterator<CasterOp, cuda::counting_iterator<uint64_t>>;
+  using RunLengthsItT = cuda::transform_iterator<ModOp, cuda::counting_iterator<RunLengthT>>;
 
-  ItemItT d_unique_items(1000U);
-  RunLengthsItT d_run_lengths(thrust::counting_iterator<RunLengthT>(0), ModOp{});
+  ItemItT d_unique_items(cuda::counting_iterator<uint64_t>(1000U), CasterOp{});
+  RunLengthsItT d_run_lengths(cuda::counting_iterator<RunLengthT>(0), ModOp{});
```

Comment:
This is somewhat concerning, since the CUDA iterators should be mostly drop-in replacements of the Thrust ones. Why is te extra transform iterator necessary? Becasue of the `float` counter? I thought @miscco fixed that already somewhere.